functions/src/index.ts

```diff
@@ -12,13 +12,70 @@ const db = admin.database(app)
 export const helloWorld = functions.https.onRequest(async (request, response) => {
   const usersRef = db.ref("users")
   await usersRef.update({
-    alanisawesome: {
-      nickname: "Alan The Machine",
+    kilvishId1: {
+      kilvishId: "kilvishId1",
+      name: "Radhey",
+      email: "radheyit@gmail.com",
+      phone: "8983026208",
     },
-    gracehop: {
-      nickname: "Amazing Grace",
+    kilvishId2: {
+      kilvishId: "KilvishId2",
+      name: "dummy Name",
+      email: "adminji@yopmail.com",
+      phone: "8208567820",
     },
   })
-  const snapshot = await db.ref("users/alanisawesome").get()
+  console.log(request.params);
+  const snapshot = await db.ref("users/kilvishId1").get()
   response.send(snapshot.val())
 })
+
+export const verifyOtp = functions.https.onRequest(async (request, response) => {
+  const reqBody = JSON.stringify(request.body);
+  const data = JSON.parse(reqBody)['data'];
+
+  const kilvishId = data["kilvishId"];
+  const phoneOtp = data["phoneOtp"];
+  const emailOtp = data["emailOtp"];
+
+  const snapshot = await db.ref(`users/${kilvishId}`).get()
+  if (snapshot != null && snapshot.val()) {
+    if (snapshot.val()["verifyPhone"] == false && snapshot.val()["verifyEmail"] == false) {
+      if (phoneOtp == "1234" && emailOtp == "5678") {
+        response.send({ "data": { "success": true } })
+      }
+    } else {
+      if (phoneOtp == "0000" && emailOtp == "0000") {
+        response.send({ "data": { "success": true } })
+      }
+    }
+  }
+  response.send({ 'data': { "success": false } })
+})
+
+
+export const login = functions.https.onRequest(async (request, response) => {
+  const reqBody = JSON.stringify(request.body);
+  const data = JSON.parse(reqBody)['data'];
+
+  const kilvishId = data["kilvishId"];
+  const email = data["email"];
+  const phone = data["phone"];
+
+  const userInfo = { kilvishId: { "kilvishId": kilvishId, "email": email, "phone": phone, "verifyPhone": false, "verifyEmail": false } };
+
+  const snapshot = await db.ref(`users/${kilvishId}`).get()
+  if (snapshot != null && snapshot.val()) {
+    if (snapshot.val()["email"] == email && snapshot.val()["phone"] == phone) {
+      /// Logic for send OTP
+      response.send({ "data": { "success": true } })
+    } else {
+      response.send({ "data": { "success": false, "message": "Wrong email phone number please enter correct correct" } })
```

Comment:
For sending success & error, the format of `response` is different. `response.send(...)` is for successful response sending. For error, it will be `response.status(non 200).send('error message')`. In this case the status code will be 400 (bad data provided).

Reply:
I frontend, we should use the status code to figure out if response was success or failure.